src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,264 @@
+        /// </summary>
+        /// <returns> A list containing absolute directory paths for CDK deployment projects.</returns>
+        public async Task<List<string>> GetRecipeDefinitionPaths()
+        {
+            var recipeDefinitionPaths = new List<string>();
+            var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+
+            var deploymentManifestModel = await GetDeploymentManifest();
+            if (deploymentManifestModel == null || deploymentManifestModel.DeploymentManifestEntries == null)
+                return recipeDefinitionPaths;
+
+            foreach (var entry in deploymentManifestModel.DeploymentManifestEntries)
+            {
+                var saveCdkDirectoryRelativePath = entry.SaveCdkDirectoryRelativePath;
+                if (string.IsNullOrEmpty(saveCdkDirectoryRelativePath))
+                    continue;
+
+                var saveCdkDirectoryAbsolutePath = _directoryManager.GetAbsolutePath(targetApplicationDirectoryPath, saveCdkDirectoryRelativePath);
+
+                if (_directoryManager.Exists(saveCdkDirectoryAbsolutePath))
+                    recipeDefinitionPaths.Add(saveCdkDirectoryAbsolutePath);
+            }
+
+            return recipeDefinitionPaths;
+        }
+
+        /// <summary>
+        /// This method parses the deployment-manifest file into a <see cref="DeploymentManifestModel"/>
+        /// </summary>
+        public async Task<DeploymentManifestModel?> GetDeploymentManifest()
+        {
+            try
+            {
+                var deploymentManifestFilePath = GetDeploymentManifestFilePath(_targetApplicationFullPath);
+
+                if (!_fileManager.Exists(deploymentManifestFilePath))
+                    return null;
+                var manifestFilejsonString = await _fileManager.ReadAllTextAsync(deploymentManifestFilePath);
+                return JsonConvert.DeserializeObject<DeploymentManifestModel>(manifestFilejsonString);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidDeploymentManifestFileException("The Deployment Manifest File is invalid.", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method parses the <see cref="DeploymentManifestModel"/> into a string and writes it to disk.
+        /// </summary>
+        private async Task WriteDeploymentManifestFile(DeploymentManifestModel deploymentManifestModel)
+        {
+            var deploymentManifestFilePath = GetDeploymentManifestFilePath(_targetApplicationFullPath);
+            var manifestFileJsonString = JsonConvert.SerializeObject(deploymentManifestModel, new JsonSerializerSettings
+            {
+                Formatting = Formatting.Indented,
+                NullValueHandling = NullValueHandling.Ignore,
+                ContractResolver = new SerializeRecipeContractResolver()
```

Comment:
Should we change the name of this resolver class to something more generic since this is not a recipe file.

Reply:
Done